state.go

```diff
@@ -180,6 +184,81 @@ func enumActors(ctx context.Context, bg *blockGetter, ts *lchtypes.TipSet, actor
 	return nil
 }
 
+func init() {
+	latestInitCode := lbiinit.AllCodes()[len(lbiinit.AllCodes())-1]
+	if name, av, ok := actors.GetActorMetaByCode(latestInitCode); ok {
+		if name != manifest.InitKey {
+			panic(xerrors.Errorf("actor code is not init: %s", name))
+		}
+
+		if av != actorstypes.Version13 {
+			panic(xerrors.Errorf(
+				"the application is out of date with the network, please update to a later version," +
+					" or if this is the latest version file an issue to update the init actor version"))
+		}
+	}
+}
```

Comment:
@ribasushi is there a better way to do this? Ideally I'd be able to just use the latest version (or map a code/actor version to the init state struct), but looking for some alternative options here.

Some notes:
1. This could probably be in a test (there should probably be a minimal testing setup 😅)
2. More generally is the network version in the state tree somewhere such that it'd be easy enough to check that the code is out of date with the network and may/may not be accurate/safe until updated?

Reply:
> you won't be able to query stuff before a particular epoch this way. Not sure this is the right way to go about this at all

Ah, I see. So I probably still need to keep this code in place (or with a test) so as to force updating the code when a new version comes out, but I should handle the historical versions as well with some sort of large map. Not sure if this would require committing to too many abstractions from the filoz folks since I'm operating on the IPLD object so I can load it into a parallel traversing HAMT which is lower level than the abstractions they currently have.